[Term Entry] C# Strings .PadRight() (#5791)
Codecademy/docs: content/c-sharp/concepts/strings/terms/padright/padright.md

---
Title: '.PadRight()'

> Description: 'The .PadRight() method in C# adds spaces or specified characters to the right of a string to achieve a specified total length.'

Reviewer: ```suggestion
Description: 'Adds padding to the right side of a string to ensure it reaches a specified total length.'
```

Subjects:
  - 'Code Foundations'
  - 'Computer Science'
Tags:
  - 'Strings'
  - 'Characters'
  - 'Methods'
  - 'Functions'
CatalogContent:
  - 'learn-c-sharp'
  - 'paths/computer-science'
---

The **`.PadRight()`** method in C# is used to add padding characters to the right of a string to achieve a specified total length. By default, it uses spaces as the padding character, but a custom character can also be specified.

## Syntax

> ```cs

Reviewer: ```suggestion
```pseudo
```

string.PadRight(int totalWidth)

> string.PadRight(int totalWidth, char paddingChar)

Reviewer: This needs to be in the following format:

```pseudo
string.PadRight(int totalWidth)
```

Or alternatively:

```pseudo
string.PadRight(int totalWidth, char paddingChar)
```

```

- `totalWidth`: The desired total length of the string, including padding. If the specified width is less than the string's length, no padding is added.

> - `paddingChar`(Optional): The character to use for padding. Defaults to a space character (' ').

Reviewer: ```suggestion
- `paddingChar`(Optional): The character to use for padding. Defaults to a space character (`' '`).
```

> **Note**: The `.PadRight()` method does not modify the original string. It generates and returns a new string with padding applied to achieve the specified total width.

## Example

### Default Padding with Spaces

```cs

> string name = "John";
> string paddedName = name.PadRight(10);
> Console.WriteLine($"'{paddedName}'");

Reviewer: ```suggestion
using System;

class Program
{
    static void Main()
    {
        string name = "John";
        string paddedName = name.PadRight(10);
        Console.WriteLine($"'{paddedName}'");
    }
}
```

```

The output of the above code will be as follows:

```shell
'John      '
```

### Custom Padding Character

```cs
string name = "John";
string paddedName = name.PadRight(10, '-');

> Console.WriteLine($"'{paddedName}'");

Reviewer: please wrap the code within class

```

The output of the above code will be as follows:

```shell
'John------'
```

### Handling Shorter Total Width

If the specified `totalWidth` is less than the length of the string, the original string is returned:

```cs
string name = "John";
string result = name.PadRight(3);

> Console.WriteLine($"'{result}'");

Reviewer: please wrap the code within class

```

The output of the above code will be as follows:

```shell
'John'
```

## Codebyte Example

> Try out the following example to see how `.PadRight()` works:

Reviewer: ```suggestion
Run the following example to see how the `.PadRight()` method works:
```
